packages/app-core/src/ui/App/ViewContainer.tsx

```diff
@@ -51,8 +58,35 @@ export default observer(function ({
     scrollRef.current?.scrollIntoView?.({ block: 'center' })
   }, [])
 
+  useEffect(() => {
+    function handleSelectView(e: Event) {
+      if (e.target instanceof Element) {
+        if (ref?.current && ref.current.contains(e.target)) {
+          // @ts-ignore
+          getRoot(view).setFocusedViewId(view.id)
+        }
+      }
+    }
+
+    document.addEventListener('mousedown', handleSelectView)
+    document.addEventListener('keydown', handleSelectView)
+    return () => {
+      document.removeEventListener('mousedown', handleSelectView)
+      document.removeEventListener('keydown', handleSelectView)
+    }
+  }, [ref, view])
+
   return (
-    <Paper ref={ref} elevation={12} className={classes.viewContainer}>
+    <Paper
+      ref={ref}
+      elevation={12}
+      className={
+        // @ts-ignore
+        getRoot(view).focusedViewId === view.id
```

Comment:
i might prefer seeing this on the session model instead of root. i generally just feel like accessing session is better than root in the first place, and it seems sort of session-y also, though it's sort of just a taste thing

Reply:
we could also correctly type it by adding to abstractsessionmodel that way